syditest/test_api.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+
+# -*- coding: utf-8 -*-
+
+# Copyright 2019 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import requests
+import re
+import sys
+import atexit
+
+# These are standard python unit tests, but are generally intended
+# to be run with trial. Trial doesn't capture logging nicely if you
+# use python 'logging': it only works if you use Twisted's own.
+from twisted.python import log
+
+from .mailsink import MailSink
+
+try:
+    from syditest_subject.launcher import SyditestLauncher
+except ImportError:
+    print("ERROR: Couldn't import launcher")
+    print(
+        "syditest needs an identity server to test: make sure, "
+        "'syditest_subject.launcher.SyditestLauncher' is in "
+        "sys.path"
+    )
+    
+    raise
+
+launcher = None
+baseUrl = None
+
+def getOrLaunchIS():
+    global launcher
+    global baseUrl
+
+    if launcher is not None:
+        return baseUrl
+
+    launcher = SyditestLauncher()
+    baseUrl = launcher.launch()
+
+    atexit.register(destroyIS)
+
+    return baseUrl
+
+def destroyIS():
+    global launcher
+
+    if launcher is not None:
+        launcher.tearDown()
+    
+
+class IsApiTest(unittest.TestCase):
+    def setUp(self):
+        self.baseUrl = getOrLaunchIS()
+
+        self.mailSink = MailSink()
+        self.mailSink.launch()
+
+    def tearDown(self):
+        self.mailSink.tearDown()
+
+    def test_v1ping(self):
```

Comment:
I feel like the endpoint tests should be in a different file to the infrastructure used to set up and tear down the system under test? Otherwise how is this going to scale?

Reply:
I have separated these out in the later PRs